api/bag_types.h

```diff
@@ -0,0 +1,38 @@
+#ifndef BAG_TYPES_H
+#define BAG_TYPES_H
+
+#include "bag_c_types.h"
+
+#include <unordered_map>
+
+
+namespace BAG
+{
+
+constexpr auto RANK = 2;
```

Comment:
Wouldn't this be a more modern approach for BAG_NULL_ELEVATION?

Reply:
Yes, I wanted to make BAG_NULL_ELEVATION constexpr as well, but I am suspecting it might be used in C code.  If it never reaches the C interface, all the better, and I will definitely be converting it to constexpr.